syntax = "proto3";
package history;




import "inspection_sdk/model/inspection/dim.proto";

/*
ListAbnormalMetrics请求
*/
message ListAbnormalMetricsRequest {
    /*
    套件id
    */
    string pluginId = 1 ;
    /*
    巡检作业ID
    */
    string jobId = 2 ;
    /*
    主机实例ID
    */
    string instanceId = 3 ;
}

/*
ListAbnormalMetrics返回
*/
message ListAbnormalMetricsResponse {
message List {
message AbnormalMetricGroups {
message AbnormalMetrics {
    /*
    维度的值列表,与指标组维度定义的顺序保持一致
    */
    repeated string dims = 1 ;
    /*
    异常指标ID
    */
    string abnormalMetricId = 2 ;
    /*
    异常指标名
    */
    string abnormalMetricName = 3 ;
    /*
    紧急程度
    */
    string level = 4 ;
}
        
    
    /*
    指标组维度定义
    */
    repeated inspection.InspectionDim dims = 1 ;
    /*
    指标组ID
    */
    string metricGroupId = 2 ;
    /*
    指标组名
    */
    string metricGroupName = 3 ;
    /*
    该指标组的所有异常指标
    */
    AbnormalMetrics abnormalMetrics = 4 ;
}
        
    
    /*
    巡检对象实例ID
    */
    string targetId = 1 ;
    /*
    巡检日志
    */
    string log = 2 ;
    /*
    异常指标列表
    */
    AbnormalMetricGroups abnormalMetricGroups = 3 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListAbnormalMetricsApi返回
*/
message ListAbnormalMetricsResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    history.ListAbnormalMetricsResponse data = 4 ;
}

